python/version_details/version_details.py

```diff
@@ -520,6 +520,10 @@ def load_data(self, entity):
             self.clear()
             return
 
+        # Ensure project schema will be loaded. The project schema is needed
+        # to reference Version fields in various UI and other elements.
+        shotgun_globals.run_on_schema_loaded(lambda : None, entity.get("project"))
```

Comment:
@thebeeland and @gomezll Ok so it's clear now. So the particular call this comment is on, line 527 in `load_data`, is really just to make sure the schema is requested to be loaded as early as possible when a new entity is requested, for that entity's project. That is why lambda none is used since don't actually need to do anything yet.

The more critical change and actual fix to the bug is in the `_version_entity_data_refreshed` function. Before, that function would do some calls in the middle and below that relied on the schema cache being loaded, if not an exception was through and the rest of the entity setup would be skipped.

That function was changed to remove those bits and defer them to the new `_finalize_requested_version` function. 99% of the time, the schema is loaded at this point and the function is called right away, but rarely (in our case when launching from web only with no previous cache available), it is not.